include/openssl/core_numbers.h

```diff
@@ -35,7 +35,12 @@ extern "C" {
  * - a function pointer extractor function with the name OSSL_'foo'
  */
 
-/* Helper macro to create the function signature typedef and the extractor */
+/*
+ * Helper macro to create the function signature typedef and the extractor
+ * |type| is the return-type of the function, |name| is the name of the
+ * function to fetch, and |args| is a parenthesized list of parameters
+ * for the function (that is, it is |name|'s function signature).
+ */
```

Comment:
The macro name OSSL_CORE_MAKE_FUNC is not very expressive IMHO.  (It _makes_ a _function_?)

Maybe replace _MAKE_ by _DECLARE_ or _DEFINE_ and find a more specifice wording for _FUNC_?

Something like  OSSL_DECLARE_DISPATCH_FUNCTION_GETTER(type,name,args)?  (just wildly guessing)

Reply:
@mattcaswell @levitte

Reply:
I 38 character macro name...  please tell me you're joking, that's half the real estate of one line (80 columns).  Also, "GETTER" is inappropriate, as that's just half of what the macro does...  it also declares a function type.

Reply:
Regarding "FUNC", we already have that on other macros, so I think it's fitting to use the same in this macro.

Reply:
All in all, I would suggest `OSSL_CORE_MAKE_DISPATCH_FUNC` if anything.  It's still pretty long, but I'll live, and I would guess that the added `DISPATCH` in the name makes it expressive enough.  What do you think, @mspncp?

Reply:
(this isn't really the right place to discuss much further...  I see enough here now to inspire someone to make the appropriate PR, together with an added `OSSL_DISPATCH_ENTRY` as suggested above)

Reply:
> I 37 character macro name... please tell me you're joking, that's half the real estate of one line (80 columns). Also, "GETTER" is inappropriate, as that's just half of what the macro does... it also declares a function type.

As I wrote, it was only a 'wild guess', a quick example shot from the hip. I know too little about the context to propose the optimal fitting name. But any of the two proposals you made is already much better than "make func".